### PR TITLE
feat: provision user delete endpoint

### PR DESCRIPTION
**Pull request by wbaker85**
API specification for the `/api/v2private/provision/user/delete` endpoint.

### CODE REVIEW

File: contracts/priv/cloud-priv.yml [hunk 1]
```diff
@@ -937,6 +968,33 @@ components:
...
+    ProvisionUserDeleteRequest:
+      type: object
+      properties:
+        userID:
+          type: string
+          description: user id to delete
+        orgID:
+          type: string
+          description: organization id the user should be deleted from
+        newOwner:
```
Comment (SimonRichardson) [inline contracts/priv/cloud-priv.yml:980]: I think this should be `newOwnerID`.

File: src/cloud/schemas/ProvisionUserDeleteRequest.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+type: object
+properties:
+  userID:
+    type: string
+    description: user id to delete
+  orgID:
+    type: string
+    description: organization id the user should be deleted from
+  newOwner:
```
Comment (SimonRichardson) [inline src/cloud/schemas/ProvisionUserDeleteRequest.yml:9]: Again `newOwnerID`

**Review by SimonRichardson — CHANGES_REQUESTED** [review]

**Review by SimonRichardson — APPROVED** [review]

File: src/cloud/schemas/ProvisionUserDeleteRequest.yml (hunk 2 above)
Comment (wbaker85) [reply src/cloud/schemas/ProvisionUserDeleteRequest.yml]: Good idea - updated! I'll update it on the IDPE end as well.

File: src/cloud/schemas/ProvisionUserDeleteRequest.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    description: organization id the user should be deleted from
+  newOwnerID:
+    type: string
+    description: user id of the owner that user-owned resources should be transferred to, if transferring user-owned resources
+  tokenOption:
+    type: string
+    enum:
+      - transfer
+      - delete
+    description: organization id the user should be deleted from
```
Comment (saraghds) [inline src/cloud/schemas/ProvisionUserDeleteRequest.yml:17]: the description is not for `tokenOption`.

File: src/cloud/schemas/ProvisionUserDeleteRequest.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  taskOption:
+    type: string
+    enum:
+      - transfer
+      - delete
+    description: organization id the user should be deleted from
```
Comment (saraghds) [inline src/cloud/schemas/ProvisionUserDeleteRequest.yml:23]: the description is not for `taskOption`.

File: src/cloud/schemas/ProvisionUserDeleteRequest.yml (hunk 3 above)
Comment (wbaker85) [reply src/cloud/schemas/ProvisionUserDeleteRequest.yml]: It sure isn't...fixed!

File: src/cloud/schemas/ProvisionUserDeleteRequest.yml (hunk 4 above)
Comment (wbaker85) [reply src/cloud/schemas/ProvisionUserDeleteRequest.yml]: Ditto to ^^

**Review by saraghds — APPROVED** [review]

File: src/cloud/schemas/ProvisionUserDeleteRequest.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  newOwnerID:
+    type: string
+    description: user id of the owner that user-owned resources should be transferred to, if transferring user-owned resources
+  tokenOption:
+    type: string
+    enum:
+      - transfer
+      - delete
+    description: delete or transfer tokens
+  taskOption:
```
Comment (StacieClark) [inline src/cloud/schemas/ProvisionUserDeleteRequest.yml:18]: Missing taskOption defaults to transfer

File: src/cloud/schemas/ProvisionUserDeleteRequest.yml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  userID:
+    type: string
+    description: user id to delete
+  orgID:
+    type: string
+    description: organization id the user should be deleted from
+  newOwnerID:
+    type: string
+    description: user id of the owner that user-owned resources should be transferred to, if transferring user-owned resources
+  tokenOption:
```
Comment (StacieClark) [inline src/cloud/schemas/ProvisionUserDeleteRequest.yml:12]: A missing tokenOption defaults to transfer

File: src/cloud/schemas/ProvisionUserDeleteRequest.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
+type: object
+properties:
+  userID:
+    type: string
+    description: user id to delete
+  orgID:
+    type: string
+    description: organization id the user should be deleted from
+  newOwnerID:
```
Comment (StacieClark) [inline src/cloud/schemas/ProvisionUserDeleteRequest.yml:9]: newOwnerID can be missing or "0" for delete options

File: src/cloud/paths/provision_user_delete.yml [hunk 8]
```diff
@@ -0,0 +1,30 @@
+post:
+  operationId: PostProvisionUserDelete
+  tags:
+    - ProvisionUserDelete
+  description: Delete a user from an organization. Deletes the user if they are part of no other organization. Idempotent.
+  requestBody:
+    description: User ID to delete
```
Comment (StacieClark) [inline src/cloud/paths/provision_user_delete.yml:7]: The description tripped me up a little as it's not quite accurate. Maybe something like "The request includes the User ID and the Org ID to delete the user and whether to transfer user resources to an optional new owner"

**Review by StacieClark — CHANGES_REQUESTED** [review]

**Review by StacieClark — APPROVED** [review]

**Pull request merged**